### DISCUSSION TITLE
is there an integration with open census?

### DISCUSSION THREAD

**Discussion by oshai** [ben-manes/caffeine#1146] [Q&A] [answered]
Hi,
We are using caffeine and would like to exports stats to the google cloud platform metrics (aka stackdriver) via open census as we already integrated such exporter.
Is there already any integration of caffeine stats with open census?
If not would it be appropriate to contribute it to this repo? and any pointers for the implementation?
Based on the wiki: https://github.com/ben-manes/caffeine/wiki/Statistics
IIUC I should implement [StatsCounter](https://javadoc.io/doc/com.github.ben-manes.caffeine/caffeine/2.5.3/com/github/benmanes/caffeine/cache/stats/StatsCounter.html)?

**Comment by ben-manes** [comment, marked as answer]
I believe OpenCensus became OpenTelemetry. The support would live in their list of [instrumentatons](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md#libraries--frameworks) if you wanted to contribute one.

I think most often the expectation is to use OT as a backend with a metrics library, like Micrometer or Dropwizard Metrics, as the api that the application's logic is instrumented with. Their java agent appears to intercept framework and async calls, to propagate the context across threads. The agent used to use Caffeine and switched to its predecessor for Android compatibility, so they would be very familiar with this library and what developers should do.

**Reply by oshai** [reply]
Thanks for the reply!
After some more investigation it seems that the current version we have with pull based approach works ok, so I am not adding that at the moment.
